polkadot/zombienet-sdk-tests/tests/disabling/slashing.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Test past-session slashing when a malicious validator backs an invalid
+//! candidate and a dispute concluding in a future session. We achieve that by
+//! making some of the honest nodes go offline.
+
+use anyhow::anyhow;
+
+use crate::helpers::{assert_blocks_are_being_finalized, assert_para_throughput};
+use polkadot_primitives::{BlockNumber, CandidateHash, DisputeState, Id as ParaId, SessionIndex};
+use serde_json::json;
+use subxt::{OnlineClient, PolkadotConfig};
+use tokio::time::Duration;
+use tokio_util::time::FutureExt;
+use zombienet_sdk::NetworkConfigBuilder;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn dispute_past_session_slashing() -> Result<(), anyhow::Error> {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+
+	let images = zombienet_sdk::environment::get_images_from_env();
+
+	let config = NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			r.with_chain("westend-local")
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				.with_default_args(vec![
+					"--no-hardware-benchmarks".into(),
+					"-lparachain=debug,runtime=debug".into(),
+				])
+				.with_genesis_overrides(json!({
+					"patch": {
+						"configuration": {
+							"config": {
+								"scheduler_params": {
+									"group_rotation_frequency": 3,
+									"max_validators_per_core": 1,
+									"lookahead": 3,
+								},
+								"needed_approvals": 2
+							}
+						}
+					}
+				}))
+				.with_node(|node| node.with_name("honest-validator-0"))
+				.with_node(|node| node.with_name("honest-validator-1"))
+				.with_node(|node| node.with_name("honest-flaky-validator-0"))
+				.with_node(|node| {
+					node.with_name("malicious-backer")
+						.with_image(
+							std::env::var("MALUS_IMAGE")
+								.unwrap_or("docker.io/paritypr/malus".to_string())
+								.as_str(),
+						)
+						.with_command("malus")
+						.with_subcommand("suggest-garbage-candidate")
+						.with_args(vec![
+							"--no-hardware-benchmarks".into(),
+							"--insecure-validator-i-know-what-i-do".into(),
+							"-lMALUS=trace,parachain=debug".into(),
+						])
+				})
+		})
+		.with_parachain(|p| {
+			p.with_id(1337)
+				.with_default_command("polkadot-parachain")
+				.with_default_image(images.cumulus.as_str())
+				.with_default_args(vec![
+					"--experimental-use-slot-based".into(),
```

Comment:
just curious, any specific reason to use the slot based collator?

Reply:
i was trying to workaround for some issue in the collator, but don't even remember what it was and now it seems no longer needed 😅